components/feature/pwa/src/main/java/mozilla/components/feature/pwa/ManifestStorage.kt

```diff
@@ -25,10 +29,24 @@ class ManifestStorage(context: Context) {
      * Load a Web App Manifest for the given URL from disk.
      * If no manifest is found, null is returned.
      *
-     * @param startUrl URL of site. Should correspond to manifest's start_url.
+     * @param url URL of site. Should correspond to an url covered by the scope of a stored manifest.
+     * @param includeTimeout Whether the manifest should only be loaded if the last usage
+     *                       has not passed the defined timeout.
+     * @param currentTime timestamp in milliseconds that is used to validate timeout.
      */
-    suspend fun loadManifest(startUrl: String): WebAppManifest? = withContext(Dispatchers.IO) {
-        manifestDao.value.getManifest(startUrl)?.manifest
+    suspend fun loadManifest(
```

Comment:
Let's leave `loadManifest` for loading based on the `startUrl`. We use this for launching the shortcut and we want to maintain lookup by the primary key.

I think we should have two new functions: `loadManifestByScope`, which includes all manifests under a given scope, and `hasRecentManifest`, which just returns a boolean.

- `loadManifest` corresponds to `getManifest`
- `loadManifestByScope` corresponds to `getManifestByScope`
- `hasRecentManifest` corresponds to a new SQL method. Perhaps we can add a method that just uses `COUNT` instead of returning the manifest, then return if count is at least 1.